src/reagent/impl/component.cljs

```diff
@@ -192,9 +192,13 @@
     (fn componentWillUpdate [nextprops]
       (this-as c (.call f c c (props-argv c nextprops))))
 
+    :getSnapshotBeforeUpdate
+    (fn getSnapshotBeforeUpdate [_ _]
+      (this-as c (.call f c c)))
```

Comment:
Maybe this fn should take `oldprops` & handle them similar to `componentDidUpdate`?